scripts/myvectorbench.py

```
@@ -475,12 +488,65 @@ def bench_knn_search(container: Container, vectors: list, wp: dict) -> dict:
+        )
+    except RuntimeError as e:
+        err = str(e)
+        if "does not exist" in err and "FUNCTION" in err:
+            probe_supported = False
+        # Other errors (e.g. index not open, dim mismatch) mean rewrite IS active.
+
+    if not probe_supported:
+        print("    ⚠ MYVECTOR_IS_ANN not supported (query rewrite inactive)")
+        return {"knn_ann_qps": 0.0, "knn_ann_p50_ms": None, "knn_ann_p99_ms": None}
+
+    rng = random.Random(77)
```

Ruff (0.15.14)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)